### DISCUSSION TITLE
Ability to split a hunk

### DISCUSSION THREAD

**Discussion by myitcv** [tpope/vim-fugitive#2214] [Q&A] [answered]
I'm relatively new to the `:G` command for exploring the staged/unstaged files/hunks, but so far I'm loving it. Thank you!

My main use case for this is interactively splitting a commit into two parts:

* part 1 
* part 2 (has a parent that is part 1)

I prepare for and choose the commit I want to submit via:

```
git rebase -i $banchpoint   # $branchpoint is the commit from which the current branch has branched
# 'e' choose which commit I want to split
git rebase --soft HEAD^
```

At this point I load Vim and `:G`. This allows me to browse the staged files and hunks and unstage the bits I want to put into part 2.

I then:

```
git commit  # and write a nice commit message for part 1
git add -A  # gather all the bits I unstaged for part 2
git commit  # I generally end up doing git commit -c REBASE_HEAD to start with the 
            # commit message of the original commit I am splitting
```

This works really well, apart from when a hunk is not fine grained enough. i.e. I only want to unstage/stage part of a hunk.

I had a search around (using the term "split a hunk") and couldn't find anything in the context of fugitive.vim (magit does have something in this space) hence the question here.

Is this something that fugitive.vim could support?

**Comment by tpope** [comment, marked as answer]
Visually select the part of the hunk you want to stage and press `s` or `-`.

**Reply by myitcv** [reply]
Ah, I see where my searching failed. I didn't see the following:

> ```
> These maps are available in both the |fugitive-summary| buffer and Fugitive
> object buffers, although not all maps make sense in all buffers.  Mappings
> that operate on the file or hunk under the cursor are generally available in
> visual mode to operate on multiple files or partial hunks.
> ```


This is fantastic, thank you very much!
